tracer/src/Datadog.Trace/Agent/Api.cs

```diff
@@ -48,6 +49,7 @@ public Api(
             IDogStatsd statsd,
             Action<Dictionary<string, float>> updateSampleRates,
             bool partialFlushEnabled,
+            string processTags,
```

Comment:
Adding process tags as `Api` state only so it can be added to client-side stats doesn't feel right.

Would it makes more sense to pass the process tags string into `SendStatsAsync()`, add it to the new `SendStatsState` there, which then passes it to `SendStatsAsyncImpl()` where you are using it?